toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -660,3 +721,106 @@ func getPartUUID(device string) (uuid string, err error) {
 	uuid = strings.TrimSpace(stdout)
 	return
 }
+
+func getSectorSizeFromFile(sectorFile string) (sectorSize uint64, err error) {
+	if exists, ferr := file.PathExists(sectorFile); ferr != nil {
+		logger.Log.Errorf("Error accessing sector size file %s", sectorFile)
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("could not find the hw sector size file %s to obtain the sector size of the system", sectorFile)
+		return
+	}
+
+	fileContent, err := file.ReadLines(sectorFile)
+	if err != nil {
+		logger.Log.Errorf("Failed to read from %s: %s", sectorFile, err)
+		return
+	}
+
+	// sector file should only have one line, return error if not
+	if len(fileContent) != 1 {
+		err = fmt.Errorf("%s has more than one line", sectorFile)
+		return
+	}
+
+	sectorSize, err = strconv.ParseUint(fileContent[0], 10, 64)
+	return
+}
+
+func getSectorSize(diskDevPath string) (logicalSectorSize, physicalSectorSize uint64, err error) {
+	const (
+		diskPathPattern = "/dev/sd"
```

Comment:
I don't think we can always bet on this to be "/dev/sd". Even if that's the only use case now, it might change in the future. I'd say we need `/dev/.+` if we're doing regular expressions.

To be consistent with how we use regular expressions elsewhere, I'd keep one pre-compiled copy instead of re-compiling it every time in this function. See how `sizeAndUnitRegexp` is defined to get what I have in mind.

Reply:
Btw., if we're already using regular expressions, we could use `^/dev/(\w+)$` and use that capture group to both check, if the string is correct and to read the disk name.

Reply:
Fixed